src/storage.rs

```diff
@@ -36,16 +70,94 @@ impl Storage {
     /// # Description
     /// Closes all file descriptors
     /// # Examples
+
+    // @TODO specify more useful error type
     pub fn close(&mut self) -> Result<(), ()> {
```

Comment:
Why was the explicit `close` method created? Probably its implementation should be part of the destructor.

Reply:
It created for case when you want to close all file descriptors without dropping the storage. 
(can be removed)

Reply:
Why is it even needed? When all descriptors are closed, Storage object can't be used anyway.